### ISSUE TITLE
[Bug] Dask-on-Ray ray scheduler does not respect dask annotations / resource requirements

### ISSUE THREAD

**Issue by mikwieczorek** [ray-project/ray#23587]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core, Ray Clusters

### Issue Severity

High: It blocks me to complete my task.

### What happened + What you expected to happen

The problem is that when using Dask-on-Ray, the Ray scheduler does not respect resource requirements marked by `dask.annotate`. 
There was a features request (https://github.com/ray-project/ray/issues/21536), which was merged, but even using the latest release of Ray I cannot make it work. 

What I would expect:
Settings:
- Using Ray Cluster with specified `rayResources` per worker or using local `ray.init(resources={...}`
- Using `dask.annotate` specifying resources required to run a computation

Results:
- Ray does not schedule jobs on workers that do not meet resource requirements specified by Dask
- Job with specified requirements are scheduled only on suitable Ray workers.

Currently, even in the latest release of Ray, scheduler seems not to respect resource requirements when scheduling jobs.

### Versions / Dependencies

ray==1.11.0
dask==2022.01.1

### Reproduction script

```
import ray
from ray.util.dask import enable_dask_on_ray
import dask
import dask.array as da

ray.init(num_cpus=2, resources={'custom_resource': 1, 'other_custom_resource': 1, 'another_custom_resource': 1})
enable_dask_on_ray()

with dask.annotate(
    ray_remote_args=dict(num_cpus=2, resources={"custom_resource": 10})
):
    d_arr = da.ones(100)

with dask.annotate(ray_remote_args=dict(resources={"other_custom_resource": 10})):
    d_arr = 2 * d_arr

sum_ = d_arr.sum()

result = sum_.compute(
    ray_remote_args=dict(max_retries=5, resources={"another_custom_resource": 10}),
    optimize_graph=False,
)
print(result)
# 200 <- always returns 200. If there is not enough resources in RayCluster I would expect it to return nothing as computation is never scheduled
ray.shutdown()
```

### Anything else

I tested the reproduction script using both `ray.init(resource={...})` and  using RayCluster with worker and same `rayResources` as in local mode.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by peytondmurray** [comment]
I spent some time looking at this. On my local system I was able to verify that the dask scheduler correctly propagates resources specified through `dask.annotate` to the ray tasks generated by the scheduler, and that the dask task wrapper correctly passes those resource annotations as arguments to `ray.remote()`. I've also checked that if I pass an annotation for a resource which doesn't exist, Ray won't schedule the dask layer as a job, as expected.

@mikwieczorek Could you post your `monitor.log`?

**Comment by mikwieczorek** [comment]
@peytondmurray Thanks for your reply. Did you check on Ray 1.11.0?
After encountering the issue I upgraded Ray to 1.12.0 and the resources started to work properly with Dask annotations in the newer version.

Sorry, for not updating the issue. I believe it can be closed, as version 1.12.0 does definitely works correctly.

**Comment by peytondmurray** [comment]
@mikwieczorek Ahh, thanks for the clarification - no, I was testing with the latest release. Seems like the issue has been resolved! :rocket:
